### PR TITLE
Add types for change-case-object

### PR DESCRIPTION
**Pull request by jameschao**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/change-case-object/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Type definitions for change-case-object 2.0
+// Project: https://github.com/BinaryThumb/change-case-object#readme
+// Definitions by: James Chao <https://github.com/jameschao>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function camelCase(s: string): string;
+export function camelCase<T>(arr: ReadonlyArray<T>): ReadonlyArray<T>;
```
Comment (demensky) [inline types/change-case-object/index.d.ts]: If you pass an array of objects, the result will not be equivalent. Example ([RunKit](https://runkit.com/demensky/5de9bc1d3d6662001a19a5c8)):

```js
const changeCaseObject = require("change-case-object");

changeCaseObject.camelCase([{a_b_c: 'e_f_g'}]); // [{aBC: 'e_f_g'}]
```

I suggest this option:

```suggestion
export function camelCase(arr: ReadonlyArray<string>): ReadonlyArray<string>;
export function camelCase(arr: ReadonlyArray<object>): ReadonlyArray<object>;
```

Comment (demensky) [inline types/change-case-object/index.d.ts]: Taking a read-only array is a good idea. But why indicate this for an array that is created each time this function is called?

```suggestion
export function camelCase<T>(arr: ReadonlyArray<T>): T[];
```

**Review by demensky — CHANGES_REQUESTED** [review]
I left comments only for `camelCase`, but they are relevant for all functions.

File: types/change-case-object/index.d.ts (hunk 1 above)
Comment (jameschao) [reply types/change-case-object/index.d.ts]: Good catch @demensky. That was unintentional.

Comment (jameschao) [reply types/change-case-object/index.d.ts]: 👍

**Comment by jameschao** [comment]
Thanks for the feedback @demensky and I've fixed the issues you identified. Would you mind taking another look? Thanks.

**Review by demensky — APPROVED** [review]

**Review by demensky — APPROVED** [review]

**Comment by orta** [comment]
Looks good to me too 👍

**Comment by jameschao** [comment]
Thanks @orta! Would this automatically create and publish a new `@types/changed-case-object` package?

**Comment by orta** [comment]
yes

**Comment by jameschao** [comment]
Thanks! I'm using it in my project now.

**Pull request merged**